lib/bundix/shell_nix_context.rb

```diff
@@ -0,0 +1,28 @@
+class Bundix
+  class ShellNixContext < Struct.new(:project, :ruby, :gemfile, :lockfile, :gemset)
+    def self.from_hash(hash)
+      p, r, gf, l, gs = hash.values_at(:project, :ruby, :gemfile, :lockfile, :gemset)
+      self.new(p,r,gf,l,gs)
```

Comment:
`new(hash.values_at(*members))` is a shorter version of this.

One issue with both implementations is that it won't fail if unknown keys are being passed.

Reply:
this could be a stricter implementation:
```ruby
x = new()
hash.each_pair { |k, v| x[k] = v }
x
```

Reply:
The template context is a subset of the options from the commands. It seems overly ceremonial to explicitly pluck out the hash members needed - but the values_at could certainly go in the commandline.

What I did notice as I was rejiggering here is that nil values can make there way to the context and only get caught late in the templating phase. I've added a test that all the fields of the context are assigned real values.